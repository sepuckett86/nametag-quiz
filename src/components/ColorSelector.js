// Presentational/Functional component
// Color input to change font color
// Color input to change background color

import React from 'react';
import PropTypes from 'prop-types';

function ColorSelector({ handleChange, color, backgroundColor }) {
  return (
    <>
    <label>
      Font Color: <input name="color" type="color" value={color} onChange={handleChange}></input>
    </label>
    <label>
      Background Color: <input name="backgroundColor" type="color" value={backgroundColor} onChange={handleChange}></input>
    </label>
    </>
  );
}

ColorSelector.propTypes = {
  handleChange: PropTypes.func.isRequired,
  color: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired
};

export default ColorSelector;
