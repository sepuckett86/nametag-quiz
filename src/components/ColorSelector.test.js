import React from 'react';
import { shallow } from 'enzyme';
import ColorSelector from './ColorSelector';

describe('ColorSelector component', () => {
  it('renders App', () => {
    const wrapper = shallow(
      <ColorSelector 
        color="#000000"
        backgroundColor="#000000"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
