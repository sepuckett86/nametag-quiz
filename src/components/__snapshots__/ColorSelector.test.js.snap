// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelector component renders App 1`] = `
<Fragment>
  <label>
    Font Color: 
    <input
      name="color"
      type="color"
      value="#000000"
    />
  </label>
  <label>
    Background Color: 
    <input
      name="backgroundColor"
      type="color"
      value="#000000"
    />
  </label>
</Fragment>
`;
